src/export/ValueSetExporter.ts

```diff
@@ -184,6 +184,28 @@ export class ValueSetExporter {
     }
   }
 
+  private addConceptComposeElement(
+    freshElement: ValueSetComposeIncludeOrExclude,
+    composeList: ValueSetComposeIncludeOrExclude[]
+  ): void {
+    if (freshElement.concept?.length > 0) {
+      const matchingSystemIndex = composeList.findIndex(compose => {
+        return (
+          compose.system === freshElement.system &&
+          compose.version === freshElement.version &&
+          compose.concept?.length > 0
```

Comment:
I think we also need to check if the `compose.valueSet` matches the `freshElement.valueSet` -- because if valueSets are _also_ specified in the compose rule, then it selects the intersection of the concepts and valueSets. We don't want to add a concept that isn't supposed to be intersected to a compose rule that has an intersection (and vice-versa).